src/backend/distributed/worker/worker_drop_protocol.c

```diff
@@ -34,6 +34,7 @@
 #include "utils/fmgroids.h"
 
 PG_FUNCTION_INFO_V1(worker_drop_distributed_table);
+PG_FUNCTION_INFO_V1(worker_drop_distributed_table_only);
```

Comment:
1) It is not clear what is `table_only`, can we maybe call it `worker_drop_shell_table` ?
2) Can you please remind me why we ended up needing this UDF? What is wrong with `worker_drop_distributed_table`? 
3) Do we need `worker_drop_distributed_table`?  Or, do we still need worker_drop_distributed_table_only /  DELETE_ALL_SHARDS / DELETE_ALL_PLACEMENTS?   It feels like we shouldn't need these at all. Maybe do the other refactor first, then we can discuss again. I know you mentioned it is to get a fresh metadata, but still if we really need fresh metadata (which should not happen), we can still do this manually. My concern is whether this somehow hides some bugs or not. That's why it bugs me a little

Reply:
As we've discussed offline, we've deciced to keep both worker_drop_shell_table and worker_drop_distributed_table. worker_drop_shell_table will only shell table without deleting tuples from pg_dist_partition, pg_dist_shard and pg_dist_placement,